Lib/importlib/_bootstrap_external.py

```diff
@@ -310,7 +320,18 @@ def cache_from_source(path, debug_override=None, *, optimization=None):
         if not optimization.isalnum():
             raise ValueError('{!r} is not alphanumeric'.format(optimization))
         almost_filename = '{}.{}{}'.format(almost_filename, _OPT, optimization)
-    return _path_join(head, _PYCACHE, almost_filename + BYTECODE_SUFFIXES[0])
+    filename = almost_filename + BYTECODE_SUFFIXES[0]
+    if bytecode_path is not None:
+        if not _path_isabs(head):
+            head = _path_join(_os.getcwd(), head)
+        if head[1] == ':':
```

Comment:
Yes, it's for Windows support. And yeah, I think currently it could get things wrong on posix; I'll also add a check that the first character is alpha; since this path is definitely absolute at this point, that means it can't be a posix path.